src/backend/codegen_cpp.cpp

```diff
@@ -127,6 +127,32 @@ namespace graphit {
 
     }
 
+    void CodeGenCPP::visit(mir::ParForStmt::Ptr par_for_stmt) {
+        printIndent();
+        auto for_domain = par_for_stmt->domain;
+        auto loop_var = par_for_stmt->loopVar;
+
+        if (par_for_stmt->grain_size != 0){
+            oss << "#pragma cilk grainsize = " << par_for_stmt->grain_size;
+
+        }
+
+        oss << std::endl;
+        printIndent();
+        oss << "cilk_for( int " << loop_var << " = ";
```

Comment:
Yep. We are able to get okay numbers with nested cilk.

Reply:
My understanding is if we don't hardcode here, we won't be able to use mixed schemes? Other way is to defer to schedule, but it is probably not required for now. We shouldn't add too many options to the scheduling language.

Reply:
I think every visitor for ParForStmt should scope and unscope the labels? Just like the default MIRVisitor?
This is important because we allow adding labels to ParForStmt.

Reply:
Makes sense. Should we do that in the backend code generation too? i don't see it being used for ForStmt in the backend code generation.